File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        i = 0
        j = len(s) - 1

        while i < j:
            left = s[i]
            right = s[j]

            if not left.isalnum():
                i = i + 1
                continue
            if not right.isalnum():
                j = j - 1
                continue
            if left.lower() != right.lower():
                return False
            
            i = i + 1
            j = j - 1

        return True

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int i = 0;
        int j = s.length() - 1;
        
        while (i < j) {
            char left = s[i];
            char right = s[j];

            if (!std::isalnum(left)) {
                i++;
                continue;
            }
            if (!std::isalnum(right)) {
                j--;
                continue;
            }
            if (std::tolower(left) != std::tolower(right)) {
                return false;
            }

            i++;
            j--;
        }

        return true;
    }
};